### PR TITLE
Master test 2

### PR DESCRIPTION
**Pull request by massandharsh**
New command added

### CODE REVIEW

**Comment by massandharsh** [comment]
---

### 📝 Overall PR Feedback

The MR introduces a new DeleteFromTableCommand and updates the CommandOrchestrator to include it. While the basic structure is in place, there are several areas for improvement:

1. The DeleteFromTableCommand is not fully implemented, as it doesn't perform the actual delete operation.
2. The command verification in DeleteFromTableCommand could be more robust.
3. The CommandOrchestrator could be more flexible in how it handles new commands.
4. There's a security concern with storing an API token directly in the code.

Please address these issues before merging. Also, ensure that you've added appropriate unit tests for the new DeleteFromTableCommand and updated existing tests if necessary.

---

### ✅ Test Results Summary

While the CI pipeline has passed, it's important to note that we don't see any new tests added for the DeleteFromTableCommand. Please add comprehensive unit tests to cover the new functionality, including edge cases and error scenarios. This will help ensure the reliability and correctness of the new command implementation.

File: src/main/java/commands/CommandOrchestrator.java [hunk 1]
```diff
@@ -5,14 +5,17 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import strategy.DataExtractionStrategy;
 
 @Getter
 @Setter
 public class CommandOrchestrator {
     private final List<Commands<?>> commandsList;
```
Comment (massandharsh) [inline src/main/java/commands/CommandOrchestrator.java:13]: 🏛️ **Category:** _Architecture_

Consider using a more flexible approach for adding commands. Instead of hardcoding the new command in the constructor, you could create a method to register commands dynamically. This would make the class more extensible for future command additions.

💡 Tip: Implement a `registerCommand(Commands<?> command)` method to add commands dynamically.

You can vote:
- [👍 Upvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6MTA5LCJyZXBvIjoic3FsLWxpdGUiLCJvd25lciI6Im1hc3NhbmRoYXJzaCIsImNvbW1pdF9pZCI6IjgwOGFhMGE5MTM4ZWZmODBmNDQ1YzhjZmE1ZDZlNTNkMGY3MTk2OGQiLCJwYXRoIjoic3JjL21haW4vamF2YS9jb21tYW5kcy9Db21tYW5kT3JjaGVzdHJhdG9yLmphdmEiLCJsaW5lIjoxMywic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJBcmNoaXRlY3R1cmUiLCJ0eXBlIjoiVVAiLCJjb21tZW50IjoiQ29uc2lkZXIgdXNpbmcgYSBtb3JlIGZsZXhpYmxlIGFwcHJvYWNoIGZvciBhZGRpbmcgY29tbWFuZHMuIEluc3RlYWQgb2YgaGFyZGNvZGluZyB0aGUgbmV3IGNvbW1hbmQgaW4gdGhlIGNvbnN0cnVjdG9yLCB5b3UgY291bGQgY3JlYXRlIGEgbWV0aG9kIHRvIHJlZ2lzdGVyIGNvbW1hbmRzIGR5bmFtaWNhbGx5LiBUaGlzIHdvdWxkIG1ha2UgdGhlIGNsYXNzIG1vcmUgZXh0ZW5zaWJsZSBmb3IgZnV0dXJlIGNvbW1hbmQgYWRkaXRpb25zLlxuXG5cdUQ4M0RcdURDQTEgVGlwOiBJbXBsZW1lbnQgYSBgcmVnaXN0ZXJDb21tYW5kKENvbW1hbmRzPD8-IGNvbW1hbmQpYCBtZXRob2QgdG8gYWRkIGNvbW1hbmRzIGR5bmFtaWNhbGx5LiIsImV4cCI6MTc1MzI1MDQ2Mn0.Mkzh5eLyoR44IKU2Ovtay3R6XrkqIkZyFj9BlRZzSAo)
- [👎 Downvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6MTA5LCJyZXBvIjoic3FsLWxpdGUiLCJvd25lciI6Im1hc3NhbmRoYXJzaCIsImNvbW1pdF9pZCI6IjgwOGFhMGE5MTM4ZWZmODBmNDQ1YzhjZmE1ZDZlNTNkMGY3MTk2OGQiLCJwYXRoIjoic3JjL21haW4vamF2YS9jb21tYW5kcy9Db21tYW5kT3JjaGVzdHJhdG9yLmphdmEiLCJsaW5lIjoxMywic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJBcmNoaXRlY3R1cmUiLCJ0eXBlIjoiRE9XTiIsImNvbW1lbnQiOiJDb25zaWRlciB1c2luZyBhIG1vcmUgZmxleGlibGUgYXBwcm9hY2ggZm9yIGFkZGluZyBjb21tYW5kcy4gSW5zdGVhZCBvZiBoYXJkY29kaW5nIHRoZSBuZXcgY29tbWFuZCBpbiB0aGUgY29uc3RydWN0b3IsIHlvdSBjb3VsZCBjcmVhdGUgYSBtZXRob2QgdG8gcmVnaXN0ZXIgY29tbWFuZHMgZHluYW1pY2FsbHkuIFRoaXMgd291bGQgbWFrZSB0aGUgY2xhc3MgbW9yZSBleHRlbnNpYmxlIGZvciBmdXR1cmUgY29tbWFuZCBhZGRpdGlvbnMuXG5cblx1RDgzRFx1RENBMSBUaXA6IEltcGxlbWVudCBhIGByZWdpc3RlckNvbW1hbmQoQ29tbWFuZHM8Pz4gY29tbWFuZClgIG1ldGhvZCB0byBhZGQgY29tbWFuZHMgZHluYW1pY2FsbHkuIiwiZXhwIjoxNzUzMjUwNDYyfQ.JV4MrSocjZYfqiNH5qbEukDspVDYOZ0BBtK4Kk_gygY)

File: src/main/java/commands/DeleteFromTableCommand.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    private final DataExtractionStrategy dataExtractionStrategy;
+
+    public DeleteFromTableCommand(DataExtractionStrategy dataExtractionStrategy) {
+        this.dataExtractionStrategy = dataExtractionStrategy;
+    }
+
+    @Override
+    public boolean verifyCommand(String command) {
+        String[] commands = commandExtractor(command);
+        return commands.length >= 5 && command.toLowerCase().contains("delete") && command.toLowerCase().contains("from") && command.toLowerCase().contains("where");
```
Comment (massandharsh) [inline src/main/java/commands/DeleteFromTableCommand.java:24]: 🛡️ **Category:** _Security_

The command verification could be more robust. Consider using a regular expression to match the exact command structure instead of just checking for the presence of certain words.

💡 Tip: Use a regex pattern like `^DELETE\s+FROM\s+\w+\s+WHERE\s+\w+\s*=\s*\w+$` for more precise command validation.

You can vote:
- [👍 Upvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6MTA5LCJyZXBvIjoic3FsLWxpdGUiLCJvd25lciI6Im1hc3NhbmRoYXJzaCIsImNvbW1pdF9pZCI6IjgwOGFhMGE5MTM4ZWZmODBmNDQ1YzhjZmE1ZDZlNTNkMGY3MTk2OGQiLCJwYXRoIjoic3JjL21haW4vamF2YS9jb21tYW5kcy9EZWxldGVGcm9tVGFibGVDb21tYW5kLmphdmEiLCJsaW5lIjoyNCwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJTZWN1cml0eSIsInR5cGUiOiJVUCIsImNvbW1lbnQiOiJUaGUgY29tbWFuZCB2ZXJpZmljYXRpb24gY291bGQgYmUgbW9yZSByb2J1c3QuIENvbnNpZGVyIHVzaW5nIGEgcmVndWxhciBleHByZXNzaW9uIHRvIG1hdGNoIHRoZSBleGFjdCBjb21tYW5kIHN0cnVjdHVyZSBpbnN0ZWFkIG9mIGp1c3QgY2hlY2tpbmcgZm9yIHRoZSBwcmVzZW5jZSBvZiBjZXJ0YWluIHdvcmRzLlxuXG5cdUQ4M0RcdURDQTEgVGlwOiBVc2UgYSByZWdleCBwYXR0ZXJuIGxpa2UgYF5ERUxFVEVcXHMrRlJPTVxccytcXHcrXFxzK1dIRVJFXFxzK1xcdytcXHMqPVxccypcXHcrJGAgZm9yIG1vcmUgcHJlY2lzZSBjb21tYW5kIHZhbGlkYXRpb24uIiwiZXhwIjoxNzUzMjUwNDYzfQ.j0JBpUlBRQTuP42XP9HWaZsKUNN42bDyb57gFjmV420)
- [👎 Downvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6MTA5LCJyZXBvIjoic3FsLWxpdGUiLCJvd25lciI6Im1hc3NhbmRoYXJzaCIsImNvbW1pdF9pZCI6IjgwOGFhMGE5MTM4ZWZmODBmNDQ1YzhjZmE1ZDZlNTNkMGY3MTk2OGQiLCJwYXRoIjoic3JjL21haW4vamF2YS9jb21tYW5kcy9EZWxldGVGcm9tVGFibGVDb21tYW5kLmphdmEiLCJsaW5lIjoyNCwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJTZWN1cml0eSIsInR5cGUiOiJET1dOIiwiY29tbWVudCI6IlRoZSBjb21tYW5kIHZlcmlmaWNhdGlvbiBjb3VsZCBiZSBtb3JlIHJvYnVzdC4gQ29uc2lkZXIgdXNpbmcgYSByZWd1bGFyIGV4cHJlc3Npb24gdG8gbWF0Y2ggdGhlIGV4YWN0IGNvbW1hbmQgc3RydWN0dXJlIGluc3RlYWQgb2YganVzdCBjaGVja2luZyBmb3IgdGhlIHByZXNlbmNlIG9mIGNlcnRhaW4gd29yZHMuXG5cblx1RDgzRFx1RENBMSBUaXA6IFVzZSBhIHJlZ2V4IHBhdHRlcm4gbGlrZSBgXkRFTEVURVxccytGUk9NXFxzK1xcdytcXHMrV0hFUkVcXHMrXFx3K1xccyo9XFxzKlxcdyskYCBmb3IgbW9yZSBwcmVjaXNlIGNvbW1hbmQgdmFsaWRhdGlvbi4iLCJleHAiOjE3NTMyNTA0NjN9.jTAb9ONp_6ifmvv6sTJX1CZfDpke5ngDC6ABYb8-uzM)

File: src/main/java/commands/DeleteFromTableCommand.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        List<RowData> filteredRows = tableRowContent.stream()
+                .filter(rowDatum -> {
+                    int colIndex = indexOfColumn(rowDatum, conditionCol);
+                    return colIndex >= 0 && !rowDatum.getData().get(colIndex).equalsIgnoreCase(conditionVal);
+                })
+                .collect(Collectors.toList());
+
+        // Here you would implement the logic to save the filteredRows back to the database
+        // For now, we will just return a success message
```
Comment (massandharsh) [inline src/main/java/commands/DeleteFromTableCommand.java:48]: 🧹 **Category:** _Code Smell_

The method currently returns a static success message without actually performing the delete operation. Implement the actual delete logic and consider returning the number of rows affected for better feedback.

💡 Tip: Implement the actual delete operation and return a message like 'X rows deleted successfully' where X is the number of affected rows.

You can vote:
- [👍 Upvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6MTA5LCJyZXBvIjoic3FsLWxpdGUiLCJvd25lciI6Im1hc3NhbmRoYXJzaCIsImNvbW1pdF9pZCI6IjgwOGFhMGE5MTM4ZWZmODBmNDQ1YzhjZmE1ZDZlNTNkMGY3MTk2OGQiLCJwYXRoIjoic3JjL21haW4vamF2YS9jb21tYW5kcy9EZWxldGVGcm9tVGFibGVDb21tYW5kLmphdmEiLCJsaW5lIjo0OCwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJDb2RlIFNtZWxsIiwidHlwZSI6IlVQIiwiY29tbWVudCI6IlRoZSBtZXRob2QgY3VycmVudGx5IHJldHVybnMgYSBzdGF0aWMgc3VjY2VzcyBtZXNzYWdlIHdpdGhvdXQgYWN0dWFsbHkgcGVyZm9ybWluZyB0aGUgZGVsZXRlIG9wZXJhdGlvbi4gSW1wbGVtZW50IHRoZSBhY3R1YWwgZGVsZXRlIGxvZ2ljIGFuZCBjb25zaWRlciByZXR1cm5pbmcgdGhlIG51bWJlciBvZiByb3dzIGFmZmVjdGVkIGZvciBiZXR0ZXIgZmVlZGJhY2suXG5cblx1RDgzRFx1RENBMSBUaXA6IEltcGxlbWVudCB0aGUgYWN0dWFsIGRlbGV0ZSBvcGVyYXRpb24gYW5kIHJldHVybiBhIG1lc3NhZ2UgbGlrZSAnWCByb3dzIGRlbGV0ZWQgc3VjY2Vzc2Z1bGx5JyB3aGVyZSBYIGlzIHRoZSBudW1iZXIgb2YgYWZmZWN0ZWQgcm93cy4iLCJleHAiOjE3NTMyNTA0NjR9.1llbFiBb88mGZBh4ILtkNZk63tqWbT3MSNosE-_9kNM)
- [👎 Downvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6MTA5LCJyZXBvIjoic3FsLWxpdGUiLCJvd25lciI6Im1hc3NhbmRoYXJzaCIsImNvbW1pdF9pZCI6IjgwOGFhMGE5MTM4ZWZmODBmNDQ1YzhjZmE1ZDZlNTNkMGY3MTk2OGQiLCJwYXRoIjoic3JjL21haW4vamF2YS9jb21tYW5kcy9EZWxldGVGcm9tVGFibGVDb21tYW5kLmphdmEiLCJsaW5lIjo0OCwic2lkZSI6IlJJR0hUIiwiY2F0ZWdvcnkiOiJDb2RlIFNtZWxsIiwidHlwZSI6IkRPV04iLCJjb21tZW50IjoiVGhlIG1ldGhvZCBjdXJyZW50bHkgcmV0dXJucyBhIHN0YXRpYyBzdWNjZXNzIG1lc3NhZ2Ugd2l0aG91dCBhY3R1YWxseSBwZXJmb3JtaW5nIHRoZSBkZWxldGUgb3BlcmF0aW9uLiBJbXBsZW1lbnQgdGhlIGFjdHVhbCBkZWxldGUgbG9naWMgYW5kIGNvbnNpZGVyIHJldHVybmluZyB0aGUgbnVtYmVyIG9mIHJvd3MgYWZmZWN0ZWQgZm9yIGJldHRlciBmZWVkYmFjay5cblxuXHVEODNEXHVEQ0ExIFRpcDogSW1wbGVtZW50IHRoZSBhY3R1YWwgZGVsZXRlIG9wZXJhdGlvbiBhbmQgcmV0dXJuIGEgbWVzc2FnZSBsaWtlICdYIHJvd3MgZGVsZXRlZCBzdWNjZXNzZnVsbHknIHdoZXJlIFggaXMgdGhlIG51bWJlciBvZiBhZmZlY3RlZCByb3dzLiIsImV4cCI6MTc1MzI1MDQ2NH0.yK2WNtZbwsHfBB_Q0ZTy8vgmywMKzvfXKqWQR8LnPs4)

File: src/main/java/commands/MultipleColumnSelectCommand.java [hunk 4]
```diff
@@ -67,6 +67,6 @@ public String invoke(String command) throws Exception {
                 )
                 .collect(Collectors.joining("\n"));
     }
-    private static final String SECRET_API_TOKEN = "opa_1234"; //Token for adding key
+    private static final String SECRET_API_TOKEN = "opaw_1234"; //Token for adding key
```
Comment (massandharsh) [inline src/main/java/commands/MultipleColumnSelectCommand.java:70]: 🛡️ **Category:** _Security_

Storing API tokens or secrets in the source code is a security risk. Consider using environment variables or a secure configuration management system to store sensitive information.

💡 Tip: Use environment variables or a secure secret management system to store and retrieve the API token.

You can vote:
- [👍 Upvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6MTA5LCJyZXBvIjoic3FsLWxpdGUiLCJvd25lciI6Im1hc3NhbmRoYXJzaCIsImNvbW1pdF9pZCI6IjgwOGFhMGE5MTM4ZWZmODBmNDQ1YzhjZmE1ZDZlNTNkMGY3MTk2OGQiLCJwYXRoIjoic3JjL21haW4vamF2YS9jb21tYW5kcy9NdWx0aXBsZUNvbHVtblNlbGVjdENvbW1hbmQuamF2YSIsImxpbmUiOjcwLCJzaWRlIjoiUklHSFQiLCJjYXRlZ29yeSI6IlNlY3VyaXR5IiwidHlwZSI6IlVQIiwiY29tbWVudCI6IlN0b3JpbmcgQVBJIHRva2VucyBvciBzZWNyZXRzIGluIHRoZSBzb3VyY2UgY29kZSBpcyBhIHNlY3VyaXR5IHJpc2suIENvbnNpZGVyIHVzaW5nIGVudmlyb25tZW50IHZhcmlhYmxlcyBvciBhIHNlY3VyZSBjb25maWd1cmF0aW9uIG1hbmFnZW1lbnQgc3lzdGVtIHRvIHN0b3JlIHNlbnNpdGl2ZSBpbmZvcm1hdGlvbi5cblxuXHVEODNEXHVEQ0ExIFRpcDogVXNlIGVudmlyb25tZW50IHZhcmlhYmxlcyBvciBhIHNlY3VyZSBzZWNyZXQgbWFuYWdlbWVudCBzeXN0ZW0gdG8gc3RvcmUgYW5kIHJldHJpZXZlIHRoZSBBUEkgdG9rZW4uIiwiZXhwIjoxNzUzMjUwNDY1fQ.yHJ2ndkF1A4pX6hlWX0yFEjzxb00_8-l6dAdAynHYOE)
- [👎 Downvote](http://localhost:4000/vote/?t=eyJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Im1hc3NhbmRoYXJzaCIsInByX251bWJlciI6MTA5LCJyZXBvIjoic3FsLWxpdGUiLCJvd25lciI6Im1hc3NhbmRoYXJzaCIsImNvbW1pdF9pZCI6IjgwOGFhMGE5MTM4ZWZmODBmNDQ1YzhjZmE1ZDZlNTNkMGY3MTk2OGQiLCJwYXRoIjoic3JjL21haW4vamF2YS9jb21tYW5kcy9NdWx0aXBsZUNvbHVtblNlbGVjdENvbW1hbmQuamF2YSIsImxpbmUiOjcwLCJzaWRlIjoiUklHSFQiLCJjYXRlZ29yeSI6IlNlY3VyaXR5IiwidHlwZSI6IkRPV04iLCJjb21tZW50IjoiU3RvcmluZyBBUEkgdG9rZW5zIG9yIHNlY3JldHMgaW4gdGhlIHNvdXJjZSBjb2RlIGlzIGEgc2VjdXJpdHkgcmlzay4gQ29uc2lkZXIgdXNpbmcgZW52aXJvbm1lbnQgdmFyaWFibGVzIG9yIGEgc2VjdXJlIGNvbmZpZ3VyYXRpb24gbWFuYWdlbWVudCBzeXN0ZW0gdG8gc3RvcmUgc2Vuc2l0aXZlIGluZm9ybWF0aW9uLlxuXG5cdUQ4M0RcdURDQTEgVGlwOiBVc2UgZW52aXJvbm1lbnQgdmFyaWFibGVzIG9yIGEgc2VjdXJlIHNlY3JldCBtYW5hZ2VtZW50IHN5c3RlbSB0byBzdG9yZSBhbmQgcmV0cmlldmUgdGhlIEFQSSB0b2tlbi4iLCJleHAiOjE3NTMyNTA0NjV9.8WDsMWy1FQlnhLoXHCb_gR3iFuaSxnUxgLX9V9mckCM)
